### PR TITLE
Setting up Spark Application with Gradle

### PR DESCRIPTION
**Pull request by leonanluppi**

### CODE REVIEW

**Comment by leonanluppi** [comment]
Hope help someone :-)

**Comment by ftomassetti** [comment]
Thanks for submitting this one!

**Comment by tipsy** [comment]
@leonanluppi , Sorry I didn't notice this before. I'll go through it later this week.

**Comment by leonanluppi** [comment]
@tipsy no worries :)

File: _posts/2015-08-23-setting-up-spark-with-gradle.markdown [hunk 1]
```diff
@@ -0,0 +1,62 @@
+---
+layout: post
+title: "Setting up Spark with Gradle"
+author: <a href="https://www.facebook.com/leonan.luppi" target="_blank">Leonan Luppi</a>
+date: 2015-08-23 10:24:03
+comments: true
+summary: >
+ We will learn how to set up a Spark Application using Gradle. For that we will use Eclipse Luna IDEA and Gradle 2.5.
+ --
```
Comment (tipsy) [inline _posts/2015-08-23-setting-up-spark-with-gradle.markdown]: Should be --- (three dashes in stead of two)

File: _posts/2015-08-23-setting-up-spark-with-gradle.markdown [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+##Gradle
+ Gradle is a build automation tool build on top of concepts of Maven and Ant using groovy language instead of XML language. It can be use on Java, Groovy and Scala project, setting up tasks, dependencies and build order.
+
+##Installation
+First we need install gradle IDEA on Eclipse, so let's open our Eclipse IDEA:
```
Comment (tipsy) [inline _posts/2015-08-23-setting-up-spark-with-gradle.markdown]: Should probably be "IDE" and not "IDEA"? (IDEA is IntelliJ)

File: _posts/2015-08-23-setting-up-spark-with-gradle.markdown [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+<br>Put a "Project Name" then select Java Quickstart in "Sample Project"
+<img src="/img/posts/settingup-spark-gradle/4.png" alt="">
+
+<br>Look a familiar file struct, but instead of "pom.xml" now you have "build.gradle" let's open your "build.gradle"
+<img src="/img/posts/settingup-spark-gradle/5.png" alt="">
+
+<br>We have a "dependencies{}" object, we need to add spark-core dependencies on our project. **So add: compile 'com.sparkjava:spark-core:2.2'**
+<img src="/img/posts/settingup-spark-gradle/6.png" alt="">
+
+<br>We have a "dependencies{}" object, we need to add spark-core dependencies on our project.
```
Comment (tipsy) [inline _posts/2015-08-23-setting-up-spark-with-gradle.markdown]: You already wrote this on line 32? The screenshots also look very similar. And what happened to the compile-line that was already there?

File: _posts/2015-08-23-setting-up-spark-with-gradle.markdown [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+}
+</code></pre>
+<img src="/img/posts/settingup-spark-gradle/6.png" alt="">
+
+<br>Right click select "Gradle" then "Refresh Dependencies.."
+<img src="/img/posts/settingup-spark-gradle/7.png" alt="">
+
+<br> Now we can CODE!
+
+<br> On "src/main/java" delete "org.gradle" package and add a new Class "Main.java". Put a simple hello world like...
```
Comment (tipsy) [inline _posts/2015-08-23-setting-up-spark-with-gradle.markdown:46]: HelloWorld.java (or change code example to "class Main"), at least they have to be synced :)

File: _posts/2015-08-23-setting-up-spark-with-gradle.markdown [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+<pre><code class="language-markup">
```
Comment (tipsy) [inline _posts/2015-08-23-setting-up-spark-with-gradle.markdown]: language-java

File: _posts/2015-08-23-setting-up-spark-with-gradle.markdown [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+<img src="/img/posts/settingup-spark-gradle/4.png" alt="">
+
+<br>Look a familiar file struct, but instead of "pom.xml" now you have "build.gradle" let's open your "build.gradle"
+<img src="/img/posts/settingup-spark-gradle/5.png" alt="">
+
+<br>We have a "dependencies{}" object, we need to add spark-core dependencies on our project. **So add: compile 'com.sparkjava:spark-core:2.2'**
+<img src="/img/posts/settingup-spark-gradle/6.png" alt="">
+
+<br>We have a "dependencies{}" object, we need to add spark-core dependencies on our project.
+<pre><code class="language-markup">
```
Comment (tipsy) [inline _posts/2015-08-23-setting-up-spark-with-gradle.markdown]: language-java (it's not 100% correct, but it gives okay syntax highlighting for this)

**Comment by tipsy** [comment]
@leonanluppi I was going to clean this up, but what happened at line 32-35 confused me, so I gave you code-review instead :)

If you have the time, it would be great if you could make instructions for IntelliJ too (community edition is free)

**Comment by leonanluppi** [comment]
@tipsy I'll check these little mistakes.

**Comment by leonanluppi** [comment]
@tipsy you can check now :)
@ftomassetti okay! glad to help :)

**Comment by tipsy** [comment]
@leonanluppi Thanks for your contribution :)
It's been published now, but I backdated it to group it with the Maven Tutorial.

**Comment by leonanluppi** [comment]
@tipsy ok no worries. I'll do for IntelliJ

**Comment by tipsy** [comment]
Please update the existing gradle tutorial and structure it like the Maven tutorial then :)

**Pull request merged**